test-data/unit/merge.test

```diff
@@ -0,0 +1,495 @@
+-- Test cases for AST merge (user for fine-grained incremental checking)
```

Comment:
The important thing here is that we test that identities of certain nodes are preserved across merge. `<id>` suffixes represent object identities (somewhat similar to `id(x)`), and they only make sense within a single test case. Two things within a single test case that share an id have the same identity, and two things with different ids have different identities.